### ISSUE TITLE
`AL03` does not detect subquery expressions

### ISSUE THREAD

**Issue by mroy-seedbox** [sqlfluff/sqlfluff#6650]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This is correctly detected as an AL03 violation (duh):

```sql
SELECT
   col * 5 + 10,
   col2 * 3 + 30
FROM tbl;
```

But this is not:

```sql
SELECT
  (SELECT MAX(t.col) AS m FROM t),
  (SELECT MAX(t.col2) AS m FROM t)
FROM tbl;
```

### Expected Behaviour

An alias should be required for the subquery expression.

### Observed Behaviour

No issues is detected.

### How to reproduce

See above.

### Dialect

Any.

### Version

Any (currently using `3.3.1`).

### Configuration

Any.

Preferably with:
```
[sqlfluff:rules:aliasing.expression]
allow_scalar = False
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
